openupgrade_scripts/scripts/loyalty/16.0.1.0/post-migration.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+from odoo.tools.translate import _
+
+_deleted_xml_records = [
+    "loyalty.sale_coupon_generate_rule",
+]
+
+
+def convert_loyalty_program_rewards(env):
+    openupgrade.m2o_to_x2m(
+        env.cr, env["loyalty.program"], "loyalty_program", "reward_ids", "reward_id"
+    )
+
+
+def convert_loyalty_program_rules(env):
+    openupgrade.m2o_to_x2m(
+        env.cr, env["loyalty.program"], "loyalty_program", "rule_ids", "rule_id"
+    )
+
+
+def compute_portal_point_name(env):
+    """This is a computed field, but the _program_type_default_values method of the
+    loyalty module sets the following values in portal_point_name depending on the
+    program_type field. This is done in post so that the language context can be used."""
+    portal_point_names = {
+        "coupons": _("Coupon point(s)"),
```

Comment:
AFAIK, these translation are never going to apply, as they are not included in the standard PO/POT files (the reference is by file name).

Reply:
https://github.com/odoo/odoo/blob/16.0/addons/loyalty/i18n/es.po#L746
But they are in the standard translation files and that is how the loyalty programs are named:
https://github.com/odoo/odoo/blob/16.0/addons/loyalty/models/loyalty_program.py#L197

Reply:
But as said, translations work referring the file:

![image](https://github.com/OCA/OpenUpgrade/assets/7165771/1afded44-ae99-442a-914b-aeb9bd4f3892)

so this is not going to work.